build/build.go

```diff
@@ -47,9 +56,52 @@ func NewBuildContext(installSuffix string, buildTags []string) *build.Context {
 		),
 		ReleaseTags: build.Default.ReleaseTags,
 		CgoEnabled:  true, // detect `import "C"` to throw proper error
+
+		IsDir: func(path string) bool {
+			if strings.HasPrefix(path, gopherJSRoot+string(filepath.Separator)) {
+				path = filepath.ToSlash(path[len(gopherJSRoot):])
+				if fi, err := vfsutil.Stat(gopherjspkg.FS, path); err == nil {
```

Comment:
This condition is repeated four times, so how about creating a function?

Reply:
Sorry but I meant `strings.HasPrefix(path, gopherJSRoot+string(filepath.Separator)` or `path = filepath.ToSlash(path[len(gopherJSRoot):])`. Aren't they repeated?

Reply:
I see, thanks for clarifying.

I've considered it, but I think keeping it as is is more readable and simpler. The duplication is locally contained in the same place. Factoring it out into a helper moves the relevant code further away from where it's being used, and there's just too little code here to factor out efficiently.